### PR TITLE
Templatized build & start from TF 2.6 base image.

### PR DESCRIPTION
**Pull request by rosbo**
- Build CPU & GPU images in parallel (faster build time)
- The GPU image is no longer layers on top of the CPU image (small GPU
image, simpler because no need to install CUDA ourselves)
- Upgrade TensorFlow to 2.6

http://b/167268016

### CODE REVIEW

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -24,8 +34,10 @@ RUN sed -i "s/httpredir.debian.org/debian.uchicago.edu/" /etc/apt/sources.list &
     apt-get install -y openssh-client && \
     /tmp/clean-layer.sh
 
+# TODO(rosbo): Is this needed? Make sure gpu and stubs are properly set.
```
Comment (Philmod) [inline Dockerfile.tmpl:37]: Are you planning to close this TODO in this PR or test after?

File: Dockerfile.tmpl [hunk 2]
```diff
@@ -39,8 +51,73 @@ RUN conda config --add channels nvidia && \
...
+    git clone --recursive https://github.com/microsoft/LightGBM && \
+    cd LightGBM && \
+    git checkout tags/v$LIGHTGBM_VERSION && \
+    mkdir build && cd build && \
+    cmake -DUSE_GPU=1 -DOpenCL_LIBRARY=/usr/local/cuda/lib64/libOpenCL.so -DOpenCL_INCLUDE_DIR=/usr/local/cuda/include/ .. && \
+    make -j$(nproc) && \
+    cd /usr/local/src/LightGBM/python-package && \
+    python setup.py install --precompile && \
+    mkdir -p /etc/OpenCL/vendors && \
+    echo "libnvidia-opencl.so.1" > /etc/OpenCL/vendors/nvidia.icd && \
```
Comment (Philmod) [inline Dockerfile.tmpl:85]: Why do you have to do this again (done on line 74)?

**Review by Philmod — APPROVED** [review]

File: Dockerfile.tmpl (hunk 1 above)
Comment (rosbo) [reply Dockerfile.tmpl]: Good catch, I removed the TODO since I confirmed this is not needed.

File: Dockerfile.tmpl (hunk 2 above)
Comment (rosbo) [reply Dockerfile.tmpl]: You are right, it was there twice in the `gpu.Dockerfile` also but clearly is not required. Removed. Thanks

**Pull request merged**